openrr-teleop/src/control_nodes_config.rs

```diff
@@ -115,6 +150,192 @@ impl ControlNodesConfig {
             }
         }
 
-        nodes
+        Ok(nodes)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use arci::{
+        DummyJointTrajectoryClient, DummyLocalization, DummyMoveBase, DummyNavigation, DummySpeaker,
+    };
+    use assert_approx_eq::*;
+    use openrr_client::{
+        create_random_jacobian_ik_solver, IkSolverParameters, IkSolverWithChain,
+        OpenrrClientsConfig,
+    };
+
+    use super::*;
+    use crate::{default_move_step_angular, default_move_step_linear, default_step_duration_secs};
+
+    #[test]
+    fn test_gen() {
+        // JoyJointTeleopConfig
+        let config = JoyJointTeleopNodeConfig {
+            mode: String::from("a"),
+            joint_step: 1.0_f64,
+            step_duration_secs: default_step_duration_secs(),
+        };
+        let _joy_tconfig = JoyJointTeleopConfig {
+            client_name: String::from("b"),
+            config,
+        };
+
+        // IkNodeTeleopConfig
+        let config = IkNodeConfig {
+            mode: String::from("c"),
+            move_step_angular: default_move_step_angular(),
+            move_step_linear: default_move_step_linear(),
+            step_duration_secs: default_step_duration_secs(),
+        };
+
+        assert_approx_eq!(config.move_step_angular[0], 0.05);          // Config default value check(move_step_angular)
+        assert_approx_eq!(config.move_step_angular[1], 0.05);
+        assert_approx_eq!(config.move_step_angular[2], 0.17);
+        assert_approx_eq!(config.move_step_linear[0], 0.01);            // Config default value check(move_step_linear)
+        assert_approx_eq!(config.move_step_linear[1], 0.01);
+        assert_approx_eq!(config.move_step_linear[2], 0.01);
+        assert_approx_eq!(config.step_duration_secs, 0.1);              // Config default value check(step_duration_secs)
```

Comment:
Are these needed?
We are not testing `IkNodeConfig` here.

Reply:
I'm sorry, this block is written first so these have not removed yet.
I think removing it is good.